src/transformers/integrations/integration_utils.py

```diff
@@ -985,6 +986,12 @@ def setup(self, args, state, model):
         self._experiment_name = os.getenv("MLFLOW_EXPERIMENT_NAME", None)
         self._flatten_params = os.getenv("MLFLOW_FLATTEN_PARAMS", "FALSE").upper() in ENV_VARS_TRUE_VALUES
         self._run_id = os.getenv("MLFLOW_RUN_ID", None)
+        self._async_log = False
+        # "synchronous" flag is only available with mlflow version >= 2.8.0
+        # https://github.com/mlflow/mlflow/pull/9705
+        # https://github.com/mlflow/mlflow/releases/tag/v2.8.0
+        if packaging.version.parse(importlib.metadata.version("mlflow")) >= packaging.version.parse("2.8.0"):
```

Comment:
This does not seem to work when `mlflow-skinny` is used. Why can't we check using `mlflow.__version__`?

Reply:
> This does not seem to work when `mlflow-skinny` is used. Why can't we check using `mlflow.__version__`?

Nice catch! You might want to change the way it checks the version string so that it works for both.

Reply:
@cchen-dialpad submitted a fix: https://github.com/huggingface/transformers/pull/29918